kong/templates/nginx_kong.lua

```diff
@@ -65,25 +61,48 @@ upstream kong_upstream {
 }
 
 map $http_upgrade $upstream_connection {
-    default keep-alive;
+    default   keep-alive;
     websocket upgrade;
 }
 
 map $http_upgrade $upstream_upgrade {
-    default '';
+    default   '';
     websocket websocket;
 }
 
+map $http_x_forwarded_proto $upstream_x_forwarded_proto {
+    default $http_x_forwarded_proto;
+    ''      $scheme;
+}
+
+map $http_x_forwarded_host $upstream_x_forwarded_host {
+    default $http_x_forwarded_host;
+    ''      $host;
+}
+
+map $http_x_forwarded_port $upstream_x_forwarded_port {
+    default $http_x_forwarded_port;
+    ''      $server_port;
+}
```

Comment:
Any reason why we're not handling those in the Lua logic but need those maps again instead? I don't remember if there was a special case that they handle in lieu of an `else` in the Lua code?